Makefile

```diff
@@ -1,6 +1,33 @@
 include version.mk
 
-ARCH := $(shell arch)
+# Check for required dependencies
+CHECK_GO := $(shell command -v go 2> /dev/null)
+CHECK_CURL := $(shell command -v curl 2> /dev/null)
+CHECK_DOCKER := $(shell command -v docker 2> /dev/null)
+
+check-go:
+ifndef CHECK_GO
+	$(error "Go is not installed. Please install Go and retry.")
+endif
+
+check-curl:
+ifndef CHECK_CURL
+	$(error "curl is not installed. Please install curl and retry.")
+endif
+
+check-docker:
+ifndef CHECK_DOCKER
+	$(error "Docker is not installed. Please install Docker and retry.")
+endif
+
+# Targets that require the checks
+build: check-go check-curl
+build-docker: check-go check-curl check-docker
+build-docker-nc: check-go check-curl check-docker
+lint: check-go check-curl check-docker
+install-linter: check-go check-curl
```

Comment:
These seem over determined. Only install-linter needs curl, for example

Reply:
Thanks for pointing this out. Corrected them to the minimum requirements.